### ISSUE TITLE
[Accessibility] Button triggering modal missing aria-haspopup

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce-gateway-amazon-pay#326]
## Audit Information
This issue came from an audit WooCommerce had Equalize Digital complete in December 2025. I'm moving the issue out of Linear (WooA11Y-416) to this repo.

- Audit ID: WooExt-511
- Component/Page: Single Products
- Affected Element: Amazon Pay button
- URL where issue was discovered: https://apay-woo.mystagingwebsite.com/product/lightweight-wool-table/

## WCAG Compliance

- Success Criterion: 4.1.2 Name, Role, Value
- WCAG Version: 2
- Level: A
- Severity: Medium

## Issue Description

When a button opens a modal or dialog, this behavior should be communicated to screen reader users to set proper expectations. In this case, the button—this is related to the previous control—lack the aria-haspopup attribute, so users are not informed that activating the button will open a modal or dialog.

### Impacted Populations

Blind and low vision people, Screen reader users (blind/low vision and neurodivergent people)

## Recommended Fix

Add aria-haspopup="dialog" to the button so screen reader users are informed of the action when the button is activated. Additionally, include the aria-controls attribute to link the button to the ID of the modal being opened.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup  and https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

## Code Example
```
<div class="amazonpay-button-container amazonpay-button-enabled amazonpay-button-container-rows" id="maxo-button-wrapper-1764832198951"><div class="amazonpay-button-view1 amazonpay-button-view1-gold" tabindex="0"><picture class="amazonpay-button-logo"><img src="https://m.media-amazon.com/images/G/01/AmazonPay/Maxo/logo._CB452516594_.svg" alt="Amazon Pay"></picture><picture class="amazonpay-button-sandbox-logo"><img src="https://m.media-amazon.com/images/G/01/AmazonPay/Maxo/sandbox_icon._CB452516595_.svg" alt=""></picture><picture class="amazonpay-button-chevrons"><img src="https://m.media-amazon.com/images/G/01/AmazonPay/Maxo/AmazonPay_button_chevron._CB1558391205_.svg" alt=""></picture></div><div class="amazonpay-button-view2"></div><div class="amazonpay-button-view3"><picture class="amazonpay-button-microtext"><img src="https://static-na.payments-amazon.com/assets/maxo/microtext/en_US.svg" aria-label="Use your Amazon Pay Sandbox test account"></picture></div></div>
```
## Screenshots

https://drive.google.com/file/d/1NT4fCSXZppk93YQmvY_SMx1XtcBSuoVu/view?usp=drivesdk

**Comment by ChristianAF** [comment]
Hey @amberhinds, this issue is not valid for the plugin. It's related to Amazon Pay SDK. we can not handle the rendeting of the button.
